lib/internal/policy/manifest.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+//
+// {
+//   "resources": {
+//      "url": {
+//        "integrity": "sri"
+//      }
+//   }
+// }
+//
+const {
+  ERR_MANIFEST_ASSERT_INTEGRITY,
+  ERR_MANIFEST_PARSE_INTEGRITY,
+} = require('internal/errors').codes;
+const SRI = require('internal/policy/sri');
+const { SafeWeakMap } = require('internal/safe_globals');
+const crypto = require('crypto');
+const { Buffer } = require('buffer');
+const { createHash, timingSafeEqual } = crypto;
+const Hash$update = Function.call.bind(crypto.Hash.prototype.update);
+const Hash$digest = Function.call.bind(crypto.Hash.prototype.digest);
+const Buffer$toString = Function.call.bind(Buffer.prototype.toString);
+const entries = Object.entries;
+const INTEGRITIES = new SafeWeakMap();
+class Manifest {
+  constructor(obj) {
+    const integrities = {
+      __proto__: null,
+    };
+    const manifestEntries = entries(obj.resources);
+    for (var i = 0; i < manifestEntries.length; i++) {
+      const { 0: url, 1: { integrity } } = manifestEntries[i];
+      if (integrity != null) {
+        try {
+          const sri = SRI.parse(integrity);
+          integrities[url] = Object.freeze(sri);
+        } catch (e) {
+          throw new ERR_MANIFEST_PARSE_INTEGRITY(url, integrity);
+        }
+      }
+    }
+    Object.freeze(integrities);
+    INTEGRITIES.set(this, integrities);
+    Object.freeze(this);
+  }
+  assertIntegrity(url, content) {
+    const integrities = INTEGRITIES.get(this);
+    const realIntegrities = new Map();
+    if (integrities && url in integrities) {
```

Comment:
Could we have a global setting that would force anything that is loaded to go through an integrity check that would fail if no hash is provided? Users could choose between the 2 behaviors.

Reply:
right now, if no hash is provided/the resource is not in the manifest it already fails. I'm not sure I understand the request.